src/Bicep.Core/Syntax/ObjectSyntaxExtensions.cs

```diff
@@ -68,6 +68,38 @@ public static class ObjectSyntaxExtensions
             return result;
         }
 
+        public static ObjectPropertySyntax? SafeGetPropertyByNameRecursive(this ObjectSyntax syntax, string[] propertyAccesses)
```

Comment:
>string[] [](http://example.com/codeflow?start=101&length=8)

Since you're relying on count, I'd use `IList<string>` instead of `string[]`. This way the function can be called in more cases without requiring a `.ToArray()` call. (Arrays implement `IList<T>` btw.)